src/app/posts/services/post-form.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { Post } from '../../shared/interfaces/post.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class PostFormService {
+  private postDataStream = new Subject<Post>();
+  constructor() { }
+
+  public updatePostData(data: Post): void {
+    this.postDataStream.next(data);
```

Comment:
how about of 
```
this.postDataStream.next([...this.postDataStream.getCurrentValue(), data]);
```
But convert it to BehaviourSubject

Reply:
Fixed in new PR Feature/hw3, thanks for the advice!